polyply/src/build_file_parser.py

```diff
@@ -258,8 +258,44 @@ def _base_parser_geometry(tokens, _type):
 
         parameters.append(_type)
         geometry_def["parameters"] = parameters
+
+        _check_geometry_def(geometry_def, _type)
         return geometry_def
 
+def _check_geometry_def(geom_def, geom_type):
```

Comment:
I suggest an API change where this function returns a boolean rather than raising the warning

Reply:
that's actually a really good one! This will align much better with the upcoming long-term dev of polyply